cmd/agent/app/processors/thrift_processor.go

```diff
@@ -76,12 +76,12 @@ func NewThriftProcessor(
 		numProcessors: numProcessors,
 	}
 	metrics.Init(&res.metrics, mFactory, nil)
+	res.processing.Add(res.numProcessors)
```

Comment:
suggestion....

Suggestion: - Make this
```go
res.processing.Add(res.numProcessors)
go func() {
  s.processBuffer()
  s.processing.Done()
}()
```

Why? It consolidates Add() and Done() close together.

the processBuffer goroutines will immediately block as there is nothing in the server's data channel.

This also nicely removes the ability to accidentally spin up multiple sets of processors for the same ThriftProcessor value that aren't tracked in the waitgroup.

Reply:
I like the suggestion, it makes WaitGroup operation more visible. It needs to be slightly modified to account for numProcessors:
```go
	res.processing.Add(res.numProcessors)
	for i := 0; i < res.numProcessors; i++ {
		go func() {
			res.processBuffer()
			res.processing.Done()
		}()
	}
```
I'll push an update to this PR.